### PR TITLE
Avodata

### PR DESCRIPTION
**Pull request by yardenTal1**
Rename the branch to 'avodata'

### CODE REVIEW

File: projects/avodata/firstStage.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        for j in range(i + 1, num_of_professions):
+            A[row][i] = 1
+            A[row][j] = -1
+            row += 1
+
+    return A
+
+
+def generate_synthetic_c(m, i, a):
+    c = np.dot(a, m[i].transpose())
```
Comment (oferb) [inline projects/avodata/firstStage.py]: You can return directly, without an intermediate `c` variable.

Same for `solve`.

The name of the method already has the information of what is returned.

File: projects/avodata/firstStage.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    for i in range(num_of_characteristics):
+        c = c_matrix[i]
+        row_in_m = solve(A, c)
+        row_in_m = (row_in_m - np.min(row_in_m)) / np.ptp(row_in_m)
+        m[i] = row_in_m
+    return m
+
+
+
+def run_check (m):
```
Comment (oferb) [inline projects/avodata/firstStage.py]: Extra space:
`run_check ( -> run_check(`

File: projects/avodata/firstStage.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+# print("The Error matrix: ")
+# print(E_matrix)
+
+# The professions are - driver, lawyer, programmer
+# The characteristics are - social status, working with people, analytical thinking
+c_matrix = [[-3, -3, -2],
+            [-1, 2, 3],
+            [-3, -3, -2]]
+m = build_m(3, c_matrix, 3)
+print(m)
```
Comment (oferb) [inline projects/avodata/firstStage.py]: Please add a newline at the end of the file.

File: projects/avodata/firstStage.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        row_in_m = solve(A, c)
+        new_m[i] = row_in_m
+    Ax = np.dot(A, new_m.transpose())
+    errors = np.subtract(Ax, c_matrix.transpose())
+
+    return new_m, errors
+
+#
+# def find_c_and_a(m):
+#     # A function that receives a matrix m with size k*n, such that k is a characteristics of
```
Comment (oferb) [inline projects/avodata/firstStage.py]: If we leave commented code - we also explain why it's commented and when it should be uncommented.
Alternatively, if you don't need the code, you can delete it. It's always stored in git's history.

File: projects/avodata/firstStage.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import math
+
```
Comment (oferb) [inline projects/avodata/firstStage.py]: Please give some general explanation of this file, and the different matrices and their dimensions.
It's basically what you wrote for `find_c_and_a` (now commented out).

File: projects/avodata/firstStage.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    row_in_m = np.linalg.lstsq(A, c, None)[0]
+    return row_in_m
+
+def build_m(num_of_professions, c_matrix, num_of_characteristics):
+    A = generate_synthetic_a(num_of_professions)
+    m = np.zeros((num_of_characteristics, num_of_professions), dtype=np.float32)
+    for i in range(num_of_characteristics):
+        c = c_matrix[i]
+        row_in_m = solve(A, c)
+        row_in_m = (row_in_m - np.min(row_in_m)) / np.ptp(row_in_m)
```
Comment (oferb) [inline projects/avodata/firstStage.py]: `ptp` is just `max` in this case, since you subtract the minimum value, so the smallest value is always 0.
So I would just replace it with `max`.
Also, please comment before:
`// We normalize to the [0,1] range:

File: projects/avodata/firstStage.py (hunk 1 above)
Comment (yardenTal1) [reply projects/avodata/firstStage.py]: done

File: projects/avodata/firstStage.py (hunk 2 above)
Comment (yardenTal1) [reply projects/avodata/firstStage.py]: done

File: projects/avodata/firstStage.py (hunk 4 above)
Comment (yardenTal1) [reply projects/avodata/firstStage.py]: Cool. good to know :)

File: projects/avodata/firstStage.py (hunk 5 above)
Comment (yardenTal1) [reply projects/avodata/firstStage.py]: yes :)

File: projects/avodata/firstStage.py (hunk 6 above)
Comment (yardenTal1) [reply projects/avodata/firstStage.py]: If I just change to max the result is not the same. Because the smallest value is 0 just after all this action actually. So max and ptp are not the same here. 
It's possible with two rows, and i did it... 
I hope its good like this.

File: projects/avodata/firstStage.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
+import numpy as np
+import math
+
+# A program that receives a score matrix (c_matrix) for characteristics that characterize certain subjects.
+# And produces a matrix M with size k*n (such that k is a characteristics of
+# professions and n is the number of professions that are compared),
+# that shows the ratio of the properties between the different professions
```
Comment (oferb) [inline projects/avodata/firstStage.py]: I'd rephrase `ratio->relative value`. Ratio is usually for `a/b`.

File: projects/avodata/firstStage.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+# The program using the List Square algorithm, the vector c with size k*1, and a
```
Comment (oferb) [inline projects/avodata/firstStage.py]: `Square->Squares`

File: projects/avodata/firstStage.py (hunk 7 above)
Comment (yardenTal1) [reply projects/avodata/firstStage.py]: done

File: projects/avodata/firstStage.py (hunk 8 above)
Comment (yardenTal1) [reply projects/avodata/firstStage.py]: done

File: projects/avodata/firstStage.py [hunk 9]
```diff
@@ -1,11 +1,13 @@
...
+import pandas as pd
+
 
 # A program that receives a score matrix (c_matrix) for characteristics that characterize certain subjects.
 # And produces a matrix M with size k*n (such that k is a characteristics of
 # professions and n is the number of professions that are compared),
-# that shows the ratio of the properties between the different professions
-# The program using the List Square algorithm, the vector c with size k*1, and a
+# that shows the relative value of the properties between the different professions
+# The program using the List Squares algorithm, the vector c with size k*1, and a
```
Comment (oferb) [inline projects/avodata/firstStage.py]: `List->Least`

File: projects/avodata/firstStage.py [hunk 10]
```diff
@@ -72,8 +79,11 @@ def run_check(m):
 
 # The professions are - driver, lawyer, programmer
 # The characteristics are - social status, working with people, analytical thinking
-c_matrix = [[-3, -3, -2],
-            [-1, 2, 3],
-            [-3, -3, -2]]
+# c_matrix = [[-3, -3, -2],
```
Comment (oferb) [inline projects/avodata/firstStage.py]: Please delete things you don't need :)

File: projects/avodata/firstStage.py [hunk 11]
```diff
@@ -72,8 +79,11 @@ def run_check(m):
...
+#             [-1, 2, 3],
+#             [-3, -3, -2]]
+
+c_matrix = create_c_matrix_from_csv('check.csv')
```
Comment (oferb) [inline projects/avodata/firstStage.py]: How about you add a `check.csv` example so people know how to run the code?
Also, would be good to explain the structure of `check.csv`.

File: projects/avodata/firstStage.py (hunk 9 above)
Comment (yardenTal1) [reply projects/avodata/firstStage.py]: done

File: projects/avodata/firstStage.py (hunk 10 above)
Comment (yardenTal1) [reply projects/avodata/firstStage.py]: done

File: projects/avodata/firstStage.py (hunk 11 above)
Comment (yardenTal1) [reply projects/avodata/firstStage.py]: What you mean in the first sentence?

**Comment by yardenTal1** [comment]
Merged avodata python file
